library/source/rum/rumHelper.bs

```diff
@@ -31,3 +31,21 @@ function lastViewEventFilePath(instanceId as string) as string
     folderPath = trackFolderPath("rum")
     return folderPath + "/_last_view_" + instanceId
 end function
+
+' ----------------------------------------------------------------
+' Merges two associative arrays into one. If a key is shared between them,
+' the second object's value will override the first's.
```

Comment:
The `datadogContext` refers to the global attributes set by the customer. My thinking is that the associative array passed specifically for an event should override the global attributes in case of conflicts